### PR TITLE
Use `import type` for importing type declarations

### PR DESCRIPTION
**Pull request by vonagam**
Right now when I open a generated typescript declaration there are errors about imports of types, like those:
```ts
import * as _ from "../gleam.d.ts";
```
```
A declaration file cannot be imported without 'import type'. 
Did you mean to import an implementation file '../gleam.js' instead?
```

This one changes printing of import statements - if an import path ends with `.d.ts` add the type modifier.

There is a downside through - `import type` was [introduced at the start of 2020](https://devblogs.microsoft.com/typescript/announcing-typescript-3-8-beta/#type-only-imports-exports) with 3.8 release. 3.5 years have passed but I don't know what minimum version of typescript is supposed to be supported by gleam.

### CODE REVIEW

**Comment by vonagam** [comment]
Fixed formatting (otherwise all tests were green).

File: compiler-core/src/javascript/import.rs
Comment (lpil) [inline compiler-core/src/javascript/import.rs:95]: Could we pass this in as an argument or other form of configuration rather than checking the path please.

**Comment by lpil** [comment]
This is a fab change! Thank you. I've left 1 note inline

File: compiler-core/src/javascript/import.rs
Comment (vonagam) [reply compiler-core/src/javascript/import.rs:95]: Initially I thought about adding an argument but I don't know Rust or this code base so went with a change of least surface level. I'll look into converting it into an argument.

Comment (vonagam) [reply compiler-core/src/javascript/import.rs]: Added boolean `is_type` property to `Import`. Can rename if there are better suggestions.

**Comment by vonagam** [comment]
Anything else holding this back?

File: compiler-core/src/javascript/import.rs
Comment (lpil) [inline compiler-core/src/javascript/import.rs:32]: I think `is_type` is a little confusing in this context. How about this?

```rust
target: JavaScriptCodegenTarget,
```
```rust
#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub enum JavaScriptCodegenTarget {
    JavaScript,
    TypeScriptDeclarations,
}
```

Comment (vonagam) [reply compiler-core/src/javascript/import.rs]: I've changed the thing from being property of an import to an argument for `to_doc` since there should be no type imports in js and no non-type imports in ts, so it is not a property of an import instance. Also less code changes.

As for `enum` - from my perspective it makes sense to introduce the enum if you expect or already know other places where it can be useful too. And instead of `TypeScriptDeclarations` I prefer shorter `TypeScript`.

Do you still want enum after current change? If yes, can you add it (into this PR or after merging)? I feel like you will implement it more properly and consistently with code base than me.

Comment (lpil) [reply compiler-core/src/javascript/import.rs]: An enum with `TypeScriptDeclarations` please, as it's specifically that format and not regular TypeScript.

Comment (vonagam) [reply compiler-core/src/javascript/import.rs]: I've updated with the enum. Is that correct usage/placement?

**Review by lpil — APPROVED** [review]
Brilliant! Thank you very much

**Pull request merged**
